WordPress/pattern-directory
# Pattern CPT: Add basic content validation when saving a pattern

See #22 — this PR addresses some easily-automated validation of the pattern content. I'm validating at the API level, since someone could theoretically handcraft an API request to `/wp-json/wp/v2/wporg-pattern/`.

✅ The REST API's default validation will prevent any invalid JSON content.
✅ Initial validation checks that there is content, and that the content is parsed as valid blocks.
✅ Content will pass the validation if there is at least one block with non-default content.

⚠️ Does it contain JS or inline CSS?
These should be stripped out by `wp_kses` for users, but we'll want to check that when we figure out permissions in general, #30

⚠️ Is the markup minimally valid?
I'm not sure if we need to check for this - on the frontend, any content generated by the editor should be valid. But in PHP, GB's parser is very forgiving. So "invalid" code, ex: `<pThis is content`, passes as a valid paragraph via the API. When that's rendered in the frontend, it triggers the invalid content warning, so anyone looking at the block would know it's broken. Checking for this would probably require something like the block directory e2e tests.

⚠️ Does it use permitted blocks? (Core only, or an allow/deny list?)
It will only pass blocks that are currently registered, so we could unregister blocks we don't want people to use (see #35).

**To test**

1. Run the tests: `yarn test:php`

or, to manually test (in wp-admin, there's no error handling on the frontend yet)

1. Add a pattern `/wp-admin/post-new.php?post_type=wporg-pattern`
2. Save the empty content
3. You should see an error message
4. Add some more empty blocks, make sure not to edit any
5. Save again, another error message
6. Add some real content
7. Save again, it should succeed this time
